### PR TITLE
Update Sentry to v6 beta

### PR DESCRIPTION
**Pull request by ueman**
Updates Sentry to v6 beta and enables sending crash reports for "bad" requests.

### CODE REVIEW

File: bin/server.dart [hunk 1]
```diff
@@ -16,6 +16,7 @@ Future<void> main() async {
     o.release = Platform.environment['VERSION'];
```
Comment (ueman) [inline bin/server.dart:16]: These are now read by Sentry itself, though with a `SENTRY_` prefix.

File: bin/server.dart [hunk 2]
```diff
@@ -16,6 +16,7 @@ Future<void> main() async {
...
     o.environment = Platform.environment['ENVIRONMENT'];
     o.debug = !isProduction;
+    o.enablePrintBreadcrumbs = false;
```
Comment (ueman) [inline bin/server.dart:19]: There are quite a lot of print statements but they don't seem like they should be recorded.

File: lib/src/pub_client.dart [hunk 3]
```diff
@@ -11,7 +11,10 @@ import 'version.dart';
 class PubClient {
   /// Creates an instance of [PubClient] using an optional [HttpClient].
   PubClient([this.httpClient]) {
-    httpClient ??= SentryHttpClient();
+    httpClient ??= SentryHttpClient(
+      captureFailedRequests: true,
+      failedRequestStatusCodes: [SentryStatusCode.range(400, 500)],
```
Comment (ueman) [inline lib/src/pub_client.dart:16]: I'm not sure if it makes sense to capture these. You know probably better if it makes sense.

File: bin/server.dart (hunk 1 above)
Comment (bruno-garcia) [reply bin/server.dart:16]: Nice. I'd need to change how I build the docker image too so I'll leave it as-is for now

**Review by bruno-garcia — APPROVED** [review]

**Pull request merged**